src/auto-evo/AutoEvoRun.cs

```diff
@@ -268,25 +270,34 @@ public void AddExternalPopulationEffect(Species species, int constant, float coe
     /// <returns>The summary of external effects.</returns>
     public LocalizedStringBuilder MakeSummaryOfExternalEffects()
     {
-        var combinedExternalEffects = new Dictionary<Tuple<Species, string>, long>();
+        var combinedExternalEffects = new Dictionary<(Species Species, string Event, Patch? Patch), long>();
 
         foreach (var entry in ExternalEffects)
         {
-            var key = new Tuple<Species, string>(entry.Species, entry.EventType);
+            var key = (entry.Species, entry.EventType, entry.Patch);
+            var speciesPopulation = entry.Species.Population;
 
             combinedExternalEffects.TryGetValue(key, out var existingEffectAmount);
 
             combinedExternalEffects[key] = existingEffectAmount +
-                entry.Constant + (long)(entry.Species.Population * entry.Coefficient) - entry.Species.Population;
+                entry.Constant + (long)(speciesPopulation * entry.Coefficient) - speciesPopulation;
         }
 
         var builder = new LocalizedStringBuilder(300);
 
         foreach (var entry in combinedExternalEffects)
         {
-            // entry.Value is the amount, Item2 is the reason string
-            builder.Append(new LocalizedString("AUTO-EVO_POPULATION_CHANGED",
-                entry.Key.Item1.FormattedName, entry.Value, entry.Key.Item2));
+            if (entry.Key.Patch == null)
+            {
+                builder.Append(new LocalizedString("AUTO-EVO_POPULATION_CHANGED_EVERYWHERE",
+                    entry.Key.Species.FormattedName, entry.Value, entry.Key.Event));
+            }
```

Comment:
I may have totally forgotten, but could you remind me what's the case when a global population change is created?

Reply:
The player gets a global malus whenever they die out in a patch and are forced in another one.

Reply:
Oh yeah that was it. But isn't that like totally unrealistic? I think it should be more like the player should get a population penalty to one patch in this case? For example the patch they chose to move to.

Reply:
If you don't mind, I think it would be much simpler to always require a population effect to be in *some* patch. It'll simplify the code design and allow making the patch field non-nullable reducing needed checks.

Reply:
What's "malus"? I think I'd prefer an approach where it is just calculated normally and if the reduction causes negative population in the target patch, just set the population there to 0. Would that work fine or is there a problem with that?

Reply:
Oh, didn't notice that malus is only known in German, thought due to its latin origin it is also an English term. Malus is a penalty.
The problem is that currently the population penalty is applied to every patch and then let the player select a patch where the population is above 0. If we only apply the penalty to the target patch this does not work and we need to calculate whether or not the player would go extinct there with the penalty applied. If he would then the patch should not be selectable.

Reply:
My opinion is that the player should be able to move to any patch they want. And if the penalty causes their population to become 0 there, then let it be so. If the player survives to the editor they'll get a bit of population, if not and they keep on dying they need to next select a new patch where they have population. Once the player dies with no population in a patch and there is no other patches where the player population is > 0, then it's game over.

Reply:
Oh, I forgot that population=0 is a valid state! Problem resolved.